src/Sentry.Unity.Android/AndroidJavaScopeObserver.cs

```diff
@@ -14,63 +14,74 @@ public class AndroidJavaScopeObserver : ScopeObserver
 
         public override void AddBreadcrumbImpl(Breadcrumb breadcrumb)
         {
-            AndroidJNI.AttachCurrentThread();
-            using var sentry = GetSentryJava();
-            using var javaBreadcrumb = new AndroidJavaObject("io.sentry.Breadcrumb");
-            javaBreadcrumb.Set("message", breadcrumb.Message);
-            javaBreadcrumb.Set("type", breadcrumb.Type);
-            javaBreadcrumb.Set("category", breadcrumb.Category);
-            using var javaLevel = breadcrumb.Level.ToJavaSentryLevel();
-            javaBreadcrumb.Set("level", javaLevel);
-            sentry.CallStatic("addBreadcrumb", javaBreadcrumb, null);
+            SentryJniExecutor.FireAndForget(() =>
```

Comment:
I understand the issue is that we must call `DetachCurrentThread()`. Why then make the code asynchronous? That is likely to have unintended consequences elsewhere.

Howe about we just wrap all these functions in closures (which you already did) and instead of running on a separate thread, do it synchronously as it is now, but also detach in the finally block?

Reply:
I tried that but kept running into `attempting to detach while still running code` errors.

Reply:
Which, fair, we could do that for the `FireAndForget` specifically.

Reply:
Alright, I've searched a bit about the error in the original issue and found some solutions. The problem with launching a separate thread in each call is that it has a significant cost.

Instead, I can think of two (and a half) solutions:
1. roll back to running on the same thread as the caller and use `Android_JNI_SetupThread` the first time it's used (guarded by a thread-local boolean flag) & `Android_JNI_GetEnv` as needed. It's decribed [here](https://android-ndk.narkive.com/uqy1gqKj/can-t-execute-javavm-detachcurrentthread-attempting-to-detach-while-still-running-code), with the native [source code](https://github.com/ennorehling/android-sdl/blob/master/jni/SDL2/src/core/android/SDL_android.c#L429), if it's feasible to include those (probably a pain, right?)
2. add a custom worker thread we let this run on, with a blocking `wait()` call so we don't make this asynchronous.
3. number 2 for old platforms we know this breaks on, original code elsewhere.